addons/auth_totp/models/res_users.py

```diff
@@ -63,14 +63,18 @@ def _rpc_api_keys_only(self):
     def _get_session_token_fields(self):
         return super()._get_session_token_fields() | {'totp_secret'}
 
-    def _totp_check(self, code):
-        sudo = self.sudo()
-        key = base64.b32decode(sudo.totp_secret)
-        match = TOTP(key).match(code)
-        if match is None:
-            _logger.info("2FA check: FAIL for %s %r", self, sudo.login)
-            raise AccessDenied(_("Verification failed, please double-check the 6-digit code"))
-        _logger.info("2FA check: SUCCESS for %s %r", self, sudo.login)
+    def _check_credentials(self, credentials, env):
+        if credentials['type'] == 'totp':
+            self._totp_rate_limit('code_check')
```

Comment:
`_totp_rate_limit` is defined in the module `auth_totp_mail`, on which this modules does not depend